store/sqlstore/link_metadata_store.go

```diff
@@ -45,7 +45,7 @@ func (s SqlLinkMetadataStore) Save(metadata *model.LinkMetadata) store.StoreChan
 		metadata.PreSave()
 
 		err := s.GetMaster().Insert(metadata)
-		if err != nil {
+		if err != nil && !IsUniqueConstraintError(err, []string{"PRIMARY", "linkmetadata_pkey"}) {
```

Comment:
Should we make this method have `Upsert` semantics instead? Wondering if there are cases where the /new/ metadata should overwrite vs. being effectively discarded. It's also a bit weird that we return the metadata passed in as though it was saved, even though what's actually in the database might be different.

Reply:
Presumably, the new data should be the same as the old data, so I don't think we need to complicate this by making it into an upsert. That also means that returning the new data should be the same as returning the existing data, although it doesn't require an extra query to get the existing one.

Reply:
Sounds good. Wasn't sure if the data /could/ be different, but sounds like not -- thanks for confirming :)

Reply:
It could possibly be different if you're linking to a web page that returns non-deterministic results in a short period of time, but that's a rare enough case to ignore.